### PR TITLE
Switch to distribution v3

### PR DESCRIPTION
**Pull request by bacongobbler**
Replaces our current usage of docker/distribution with a more up-to-date release.

See https://github.com/oras-project/oras-go/pull/13 for a similar change being made upstream. Once upstream's change has been merged and a new release has been cut, we can drop the dependency on docker/distribution.

### CODE REVIEW

**Review by cndoit18 — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -12,6 +12,7 @@ require (
 	github.com/asaskevich/govalidator v0.0.0-20200428143746-21a406dcc535
 	github.com/containerd/containerd v1.5.0-rc.3
 	github.com/cyphar/filepath-securejoin v0.2.2
+	github.com/distribution/distribution/v3 v3.0.0-20210804104954-38ab4c606ee3
 	github.com/docker/distribution v2.7.1+incompatible
```
Comment (hickeyma) [inline go.mod:16]: Should this be removed?

**Review by hickeyma — COMMENTED** [review]
Should the docker/distribution still be referenced?

File: go.mod (hunk 1 above)
Comment (bacongobbler) [reply go.mod:16]: I mentioned this in the OP:

> Once upstream's change has been merged and a new release has been cut, we can drop the dependency on docker/distribution.

We have to wait until oras-go has merged https://github.com/oras-project/oras-go/pull/13 and cut a new release before we can drop this dependency

Comment (hickeyma) [reply go.mod:16]: Ok, sorry missed that

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @bacongobbler

**Review by technosophos — APPROVED** [review]
LGTM 2

**Pull request merged**